CMSgov/easi-app
# EASI-3857: Minio in k8s for EASi

# EASI-3857


## Changes and Description

- Introduce k8s base manifest for Minio to enable mock s3 bucket uploads
  - Minio console exposed via ingress. script will output link for ease of access
- Combine DB-migrate job into DB manifest files to simplify deployment



## How to test this change
Execute the k8s_dev script to deploy EASi to k8s via Rancher
    
```sh
./scripts/k8s_dev.sh
```
Output should mirror something similar as below:

```sh
❄️  Deploying EASi via Kustomize  ❄️
namespace/justin created
configmap/db-configmap created
configmap/easi-backend-configmap created
configmap/easi-client-configmap created
configmap/minio-mc-config created
service/db created
service/easi-backend created
service/easi-client created
service/email created
service/minio created
deployment.apps/db created
deployment.apps/easi-backend created
deployment.apps/easi-client created
deployment.apps/email created
deployment.apps/minio created
job.batch/db-migrate created
job.batch/minio-mc created
ingress.networking.k8s.io/easi-backend-ingress created
ingress.networking.k8s.io/easi-client-ingress created
ingress.networking.k8s.io/email-ingress created
ingress.networking.k8s.io/minio-console-ingress created
❄️  EASi: http://justin.localdev.me ❄️
❄️  Mailcatcher: http://justin-email.localdev.me ❄️
❄️  Minio: http://justin-minio.localdev.me ❄️
```

You should now be able to create an intake in EASi and upload a document. 🌟👍🏻🌟👍🏻🌟👍🏻🌟



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
